### DISCUSSION TITLE
Гиперссылка в описании плейлиста

### DISCUSSION THREAD

**Discussion by Bedmit** [Chimildic/goofy#86] [Идея] [answered]
Возможно ли в описании плейлиста сгенерированного по другому плейлисту (треку), вставлять ссылку этот источник?
Конкретный пример. 
```
function Fresh(values) { 
  let playlistId = Helper.parseId(values.text.playlistUrl);
  let Fresh = Source.getPlaylistTracks('', playlistId);
  Filter.replaceWithSimilar(Fresh, Source.getSavedTracks(), RecentTracks.get());
  Filter.matchExcept(Fresh, 'mix|club|radio|edit|live|karaoke');

  let playlist = Playlist.getById(playlistId);
  
  Playlist.saveAsNew({
    name: playlist.name + ' *',
    description: 'Свежий плейлист ' + playlist.name,
    public: false,
    sourceCover: playlist.images[0].url,
    tracks: Fresh,  
  });  
}

```
Хотелось чтобы вместо playlist.name в описании была ссылка на плейлист.

**Comment by Chimildic** [comment, marked as answer]
Обычный пользователь не может делать гиперссылки. Максимум ссылка как текст. Но это бесполезно по большей части.
 
Гиперссылки доступны только _проверянным_ пользователям (кураторы, модераторы, исполнители, разработчики).

**Reply by Bedmit** [reply]
Жаль. А как правильно прописать, чтобы отображалась не гиперссылка, а просто ссылка в виде текста, который можно будет скопировать из описания? Просто `playlistUrl `? 

UPD Этот вопрос отменяется. Методом научного тыка определил, что это `values.text.playlistUrl`
Конечно не так элегантно  как хотелось, но сойдет и так. 

Ан нет. Вопрос не отменяется. Ссылки получаются такого вида: `https://open.spotify.com/playlist/37i9dQZF1DWTfrr8pte1rT?si=tXPTR-qoSOutHi3EtcmA6g&utm_source=copy-link&dl_branch=1`

**Reply by Chimildic** [reply]
- У объекта плейлиста есть ссылка `playlist.external_urls.spotify`
- Поскольку есть `playlistId`, можно самим сделать ссылку 
```js
`https://open.spotify.com/playlist/${playlistId}`
```
- И третий вариант, это обрезать `playlistUrl.split('?')[0]`

**Reply by Bedmit** [reply]
Второй вариант не сработал - при сохранении скрипта выдает синтаксическую ошибку. Подчеркивает двоеточие.
Третий вариант сохраняется, но при запуске выдает ошибку.
Первый вариант работает. 

Но в итоге похоже придется отказаться от этой затеи. Если ПК копирование описания работает нормально, то на телефоне может выдавать такое `https:&#x2F;&#x2F;open.spotify.com&#x2F;playlist&#x2F;3jxMoEAFkRawkZJlY9Yuke` , а может и вообще копирование из интерфейса плейлиста быть неактивным. Ну и вообще как-то все сложно и не стабильно получается.

**Reply by Chimildic** [reply]
Второй вариант выглядит так
```js
  let playlistId = 'id'  
  let desc = 'Свежий плейлист ' + `https://open.spotify.com/playlist/${playlistId}`
  // На выходе: "Свежий плейлист https://open.spotify.com/playlist/id"
```
Третий вариант был заточен под ссылку, которую в пример привели (там знак `?` присутствует). Ошибок воспроизвести не получилось.
```js
  let playlistUrl = 'https://open.spotify.com/playlist/37i9dQZF1DWTfrr8pte1rT?si=tXPTR-qoSOutHi3EtcmA6g&utm_source=copy-link&dl_branch=1'  
  let desc = playlistUrl.split('?')[0]
```

**Reply by Bedmit** [reply]
Да, теперь все получилось. Но сделал уже чисто из спортивного интереса. На телефоне это все равно оказалось бессмысленным, т.к. в меню изменения описания плейлиста в котором планировалось копировать ссылку, все спецсимволы отображаются кодом и естественно для скрипта уже не годятся в таком виде.

![Screenshot_20210615-212152](https://user-images.githubusercontent.com/76230194/122104783-b7089380-ce20-11eb-98b4-ea9049d16d33.jpg)

**Reply by Chimildic** [reply]
Насколько вижу, отображается ссылка нормально. Только при редактировании появляется подмена. Можно просто `replace` добавить. 
```js
let playlistId = Helper.parseId(values.text.playlistUrl.replace(/&#x2F;/g, '/'));
```

**Comment by Bedmit** [comment]
И еще попутно вопрос. Можно ли записывать в кэш ссылку на плейлист?
Идея заключается в том, чтобы при запуске вышеуказанного скрипта через веб форму, если в строку не вставлена ссылка на плейлист,  скрипт отрабатывался  по последней заданной перед этим ссылке. 
Для себя вижу это только через запись ссылки в кэш и заменой ее при появлении новой. Такое возможно? Если возможно, то как это грамотно реализовать?
Красивей всего конечно было бы, чтобы ссылка сохранялась там же в веб форме. Но что-то мне подсказывает, что это малореально.

**Reply by Chimildic** [reply]
Так можно попробовать, не проверял 
```js
function Fresh(values) { 
  const filename = 'LastPlaylistFreshFunc.txt'
  let playlistId;
  if (values.text.playlistUrl.length == 0) {
    playlistId = Cache.read(filename);
  } else {
    playlistId = Helper.parseId(values.text.playlistUrl);  
    Cache.write(filename, playlistId);
  }  

  // ...
}
```

**Reply by Bedmit** [reply]
Вставил. Все отлично работает! Останавливаюсь на этом варианте. Спасибо.
